core/server/worker/src/main/java/alluxio/worker/block/DefaultBlockWorker.java

```diff
@@ -508,7 +508,14 @@ public BlockReader createUfsBlockReader(long sessionId, long blockId, long offse
       boolean positionShort, Protocol.OpenUfsBlockOptions options)
       throws BlockDoesNotExistException, IOException {
     try {
-      openUfsBlock(sessionId, blockId, options);
+      if (!options.hasUfsPath() && options.hasBlockInUfsTier() && options.getBlockInUfsTier()) {
```

Comment:
I was just looking at this today. The `options.hasBlockInUfsTier()` should not be necessary because `block_in_ufs_tier` is a boolean and the default value is `false`. Actually this logic of changing options should be moved to `createBlockReader`. Please take a look at https://github.com/Alluxio/alluxio/pull/15382 for related changes.

Reply:
We can merge your PR first so we can adopt the change.